crates/adapters/src/transport/http/input.rs

```diff
@@ -104,27 +108,32 @@ impl HttpInputEndpoint {
         self.inner.status_notifier.send_replace(());
     }
 
-    fn push_bytes(&self, bytes: &[u8]) -> Vec<ParseError> {
-        self.inner
-            .consumer
-            .lock()
-            .unwrap()
-            .as_mut()
-            .unwrap()
-            .input_fragment(bytes)
-    }
-
-    fn eoi(&self) -> Vec<ParseError> {
-        self.inner.consumer.lock().unwrap().as_mut().unwrap().eoi()
+    fn push(&self, bytes: Option<&[u8]>, errors: &mut CircularQueue<ParseError>) -> usize {
+        let mut guard = self.inner.cp.lock().unwrap();
+        let parser = &mut guard.as_mut().unwrap().1;
+        let (num_records, mut new_errors) = match bytes {
+            Some(bytes) => parser.input_fragment(bytes),
+            None => parser.end_of_fragments(),
+        };
+        self.inner.queued.fetch_add(num_records, Ordering::SeqCst);
+        parser.flush_all();
```

Comment:
I think this might be fine for the HTTP connector, but just curious: why did you choose to flush this one instantly instead of buffering like you do in other connectors.

Reply:
Hmm, there was some reason it didn't work the "usual" way when I was building this. I deferred figuring it out and then I forgot to go back to it. I'll spend a few minutes now trying again.

Reply:
It turned out it didn't need to be different anymore, so I fixed that, and then I realized that I had a lot of common code that could be a nice `InputQueue` abstraction so I added that on top too.